public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -943,6 +945,25 @@ export class LokiDatasource
     return { ...query, expr: expression };
   }
 
+  /**
+   * Implemented as part of `DataSourceWithQueryModificationSupport`. Returns a list of operation
+   * types that are supported by `modifyQuery()`.
+   */
+  getSupportedQueryModifications() {
+    return [
+      'ADD_FILTER',
+      'ADD_FILTER_OUT',
+      'ADD_LOGFMT_PARSER',
+      'ADD_JSON_PARSER',
+      'ADD_UNPACK_PARSER',
+      'ADD_NO_PIPELINE_ERROR',
+      'ADD_LEVEL_LABEL_FORMAT',
+      'ADD_LABEL_FILTER',
+      'ADD_STRING_FILTER',
+      'ADD_STRING_FILTER_OUT',
```

Comment:
I'd love to, but it will make backwards compatibility more complex because in the original `modifyQuery` implementation, the type is just a `string` https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L583

Unlike toggleable filters, which predefines the supported ops https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/logs.ts#L265

Reply:
Semi-related, created a new type for the supported actions: QueryFixType.